modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,345 @@
+/*
+By downloading, copying, installing or using the software you agree to this license.
+If you do not agree to this license, do not download, install,
+copy or use the software.
+
+
+                          License Agreement
+               For Open Source Computer Vision Library
+                       (3-clause BSD License)
+
+Copyright (C) 2000-2015, Intel Corporation, all rights reserved.
+Copyright (C) 2009-2011, Willow Garage Inc., all rights reserved.
+Copyright (C) 2009-2015, NVIDIA Corporation, all rights reserved.
+Copyright (C) 2010-2013, Advanced Micro Devices, Inc., all rights reserved.
+Copyright (C) 2015, OpenCV Foundation, all rights reserved.
+Copyright (C) 2015, Itseez Inc., all rights reserved.
+Third party copyrights are property of their respective owners.
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+  * Redistributions of source code must retain the above copyright notice,
+    this list of conditions and the following disclaimer.
+
+  * Redistributions in binary form must reproduce the above copyright notice,
+    this list of conditions and the following disclaimer in the documentation
+    and/or other materials provided with the distribution.
+
+  * Neither the names of the copyright holders nor the names of the contributors
+    may be used to endorse or promote products derived from this software
+    without specific prior written permission.
+
+This software is provided by the copyright holders and contributors "as is" and
+any express or implied warranties, including, but not limited to, the implied
+warranties of merchantability and fitness for a particular purpose are disclaimed.
+In no event shall copyright holders or contributors be liable for any direct,
+indirect, incidental, special, exemplary, or consequential damages
+(including, but not limited to, procurement of substitute goods or services;
+loss of use, data, or profits; or business interruption) however caused
+and on any theory of liability, whether in contract, strict liability,
+or tort (including negligence or otherwise) arising in any way out of
+the use of this software, even if advised of the possibility of such damage.
+*/
+
+
+/*
+This file contains implementation of the bio-inspired features (BIF) approach
+for computing image descriptors, applicable for human age estimation. For more
+details we refer to [1,2].
+
+REFERENCES
+  [1] Guo, Guodong, et al. "Human age estimation using bio-inspired features."
+      Computer Vision and Pattern Recognition, 2009. CVPR 2009.
+  [2] Spizhevoi, A. S., and A. V. Bovyrin. "Estimating human age using
+      bio-inspired features and the ranking method." Pattern Recognition and
+      Image Analysis 25.3 (2015): 547-552.
+*/
+#include "opencv2/face/face_alignment.hpp"
+#include "opencv2/objdetect.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
```

Comment:
@StevenPuttemans , in my opinion, it will be better not to incorporate image reading to the algorithm. Some time ago I tried to remove this dependency where possible: #1063 .
Maybe you can create sample training application which will read all files and call corresponding methods. And instead of:
```.cpp
void ShapePredictor :: scaleData( std::map<string,std::vector<cv::Point2f> > &_trainlandmarks,
                                std::vector<string> &_trainimages ,Size s)
```
you can have:
```.cpp
void ShapePredictor :: scaleData( std::vector<cv::Point2f> &_trainlandmarks, InputArray image, Size s)
```
Instead of:
```.cpp
void ShapePredictor :: calcMeanShape(std::map<string,std::vector<cv::Point2f> > &_trainlandmarks,std::vector<string> &_trainimages){
```
write something like:
```.cpp
void ShapePredictor :: beginMeanCalc(InputArray img)
void ShapePredictor :: endMeanCalc()
void ShapePredictor :: addImage(std::vector<cv::Point2f> &_trainlandmarks, InputArray img)
// or
void ShapePredictor :: addImages(std::vector<cv::Point2f> &_trainlandmarks, InputArrayOfArrays imgs)
```